test/expr/postfix/dot/init_ref_delegation.swift

```diff
@@ -323,12 +323,12 @@ class TestOverloadSets {
     self.init(5, 5) // expected-error{{extra argument in call}}
   }
   
-  convenience init(a : Z0) { // expected-note{{candidate has partially matching parameter list (a: Z0)}}
+  convenience init(a : Z0) { // expected-note{{candidate expects value of type 'Z0' for parameter #1}}
```

Comment:
From what I remember from the other PR, the calls that trigger those have also a relabel fix alongside the argument mismatch. I think this is fine, but do you think to mention that in some way could be valid in those situations?

Reply:
I don't really know, this is something I was thinking about as well. I don't really like `partially matching parameter list` note because it's as un-actionable as it gets but I don't know how to also mention a label here. It would be very unfortunate to produce two diagnostics/notes - one for labeling and one for type because that would be confusing...

Reply:
Yeah, I agree that show two notes wouldn't be ideal. But in the other PR I also thought that `partially matching parameter list` was not ideal, but I did choose to show the relabel diagnostic, but to be honest there was no particular reason other than is better than `partially matching parameter list`. But in fact neither correct the type or relabel the call would fix the problem by itself because both type and label are wrong in this situation. 
I've been thinking about it and also not sure how to mention the label and also don't know how much we are loosing not showing one of them ... probably not much since as you mention both are actionable

Reply:
I think we could re-visit this once it's possible to produce a fix per label mismatch, that way we could probably coalesce type and label mismatch for each argument and produce a much better note...